controllers/configurationpolicy_controller.go

```diff
@@ -561,19 +559,23 @@ func (r *ConfigurationPolicyReconciler) cleanUpChildObjects(plc policyv1.Configu
 }
 
 func (r *ConfigurationPolicyReconciler) definitionIsDeleting() (bool, error) {
-	key := types.NamespacedName{Name: "configurationpolicies.policy.open-cluster-management.io"}
-	v1def := extensionsv1.CustomResourceDefinition{}
+	crdGVR := schema.GroupVersionResource{
+		Group: "apiextensions.k8s.io", Version: "v1", Resource: "customresourcedefinitions",
+	}
+	crdName := "configurationpolicies.policy.open-cluster-management.io"
 
-	v1err := r.Get(context.TODO(), key, &v1def)
+	v1def, v1err := r.TargetK8sDynamicClient.Resource(crdGVR).Get(context.TODO(), crdName, metav1.GetOptions{})
```

Comment:
I think the CRD will be on the hosting cluster, not the target cluster.

Also, I'm wondering if going through the dynamic client removes the need for https://github.com/open-cluster-management-io/config-policy-controller/pull/88 ? Since I can't tell how the cache settings there would get to the dynamic client (but maybe I'm missing something)

Reply:
Sorry about that. I didn't mean to commit this part.

Reply:
I was originally going towards the route of using the dynamic client but then realized it did actually need to be a client pointing to the cluster the config-policy-controller is running on. I decided to not have yet another client on the reconciler and figured the caching provided by #88 might be a good thing rather than query for the CRD on every ConfigurationPolicy deletion.

The latest push removes the code that was accidentally committed.